no-literal-checker/src/main/java/org/checkerframework/checker/noliteral/NoLiteralVisitor.java

```diff
@@ -0,0 +1,259 @@
+package org.checkerframework.checker.noliteral;
+
+import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.noliteral.qual.MaybeDerivedFromConstant;
+import org.checkerframework.checker.noliteral.qual.NonConstant;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.type.AnnotatedTypeParameterBounds;
+import org.checkerframework.framework.type.TypeHierarchy;
+import org.checkerframework.framework.type.visitor.AnnotatedTypeScanner;
+import org.checkerframework.javacutil.ElementUtils;
+
+/**
+ * This visitor has a lot of code in it that permits this type system to operate as expected with
+ * the bottom type as its default. Most of that code is either from or inspired by the AWS Data
```

Comment:
Saying something about the relationship with the Nullness Checker might be useful, since it also has the bottom type as its default.  Should there be a subtype of BaseTypeVisitor for type systems with bottom as their default?

Reply:
This checker has no direct relationship with the Nullness Checker. I did add a mention of it in the manual, though, when I mention that this checker doesn't consider `null` to be a constant.

I do think there should be a subtype of `BaseTypeVisitor` for bottom-default type systems. For right now, I'm not willing to take on that problem, though.